FWCore/Services/plugins/Timing.cc

```diff
@@ -355,6 +367,48 @@ namespace edm {
           countSubProcessesPostEvent_.emplace_back(std::make_unique<std::atomic<unsigned int>>(0));
         }
       });
+      setTaskCallbacks(iRegistry);
+    }
+
+    void Timing::setTaskCallbacks(ActivityRegistry& iRegistry) {
+      iRegistry.preSourceSignal_.connect([this](auto) { addTask(); });
```

Comment:
Why `preSourceSignal_.connect()` instead of the "more standard" `watchPreSourceEvent()`?

Why register separate callbacks for most of the signals, but not for ESModule?

Reply:
> Why preSourceSignal_.connect() instead of the "more standard" watchPreSourceEvent()?

The `watch` style methods require a new member function of the Service to be written while the `connect` allow the use of lambdas.

Reply:
> Why register separate callbacks for most of the signals, but not for ESModule?

The ESModule callbacks are always used, regardless of the options chosen while many of the other ones can be switched off.

Reply:
> > Why preSourceSignal_.connect() instead of the "more standard" watchPreSourceEvent()?
> 
> The `watch` style methods require a new member function of the Service to be written while the `connect` allow the use of lambdas.

I see `watch` style functions being used with lambdas, e.g. here
https://github.com/cms-sw/cmssw/blob/a9d66506bc61716e40a095faf6883fa8795b76e0/FWCore/Services/plugins/ZombieKillerService.cc#L65-L66